src/ir/module-splitting.cpp

```diff
@@ -369,17 +429,57 @@ void ModuleSplitter::setupTablePatching() {
     }
   });
 
+  if (replacedElems.size() == 0) {
+    // No placeholders to patch out of the table
+    return;
+  }
+
+  if (tableManager.activeBase.global) {
+    assert(primary.table.segments.size() == 1 &&
+           "Unexpected number of segments with non-const base");
+    assert(secondary.table.segments.size() == 0);
+    // Since addition is not currently allowed in initializer expressions, we
+    // need to start the new secondary segment where the primary segment starts.
+    // The secondary segment will contain the same primary functions as the
+    // primary module except in positions where it needs to overwrite a
+    // placeholder function. All primary functions in the table therefore need
+    // to be imported into the second module. TODO: use better strategies here,
+    // such as using ref.func in the start function or standardizing addition in
+    // initializer expressions.
+    const Table::Segment& primarySeg = primary.table.segments.front();
+    std::vector<Name> secondaryElems;
+    secondaryElems.reserve(primarySeg.data.size());
+
+    // Copy functions from the primary segment to the secondary segment,
+    // replacing placeholders and creating new exports and imports as necessary.
+    auto replacement = replacedElems.begin();
+    for (Index i = 0;
+         i < primarySeg.data.size() && replacement != replacedElems.end();
+         ++i) {
+      if (replacement->first == i) {
+        secondaryElems.push_back(replacement->second);
```

Comment:
At this point, are `replacement->second` and `primatySeg.data[i]` different? If they are the same, I guess changing this to `primarySeg.data[i]` to be consistent with `else` clause below will make it clear that we are doing the same action.

Reply:
Yes, they are different. In the `if` case, `primarySeg.data[i]` would be a placeholder function but we want to use the original secondary function. In the `else` case, `primarySeg.data[i]` is the primary function we need to import/export and place in the secondary table.

I will add a comment about this.